src/client.rs

```diff
@@ -46,20 +71,21 @@ impl Client {
     }
 
     /// Construct a pair of encrypted shares based on the input data.
-    pub fn encode_simple(&mut self, data: &[Field]) -> Result<(Vec<u8>, Vec<u8>), EncryptError> {
-        let copy_data = |share_data: &mut [Field]| {
+    pub fn encode_simple(&mut self, data: &[F]) -> Result<(Vec<u8>, Vec<u8>), ClientError> {
+        let copy_data = |share_data: &mut [F]| {
             share_data[..].clone_from_slice(data);
         };
         self.encode_with(copy_data)
+            .or_else(|err| Err(ClientError::Encrypt(err)))
```

Comment:
```suggestion
        Ok(self.encode_with(copy_data)?)
```
Because we have `Encrypt(#[from] EncryptError)`, we get an appropriate `From<EncryptError> for ClientError` for free, [and the `?` operator will automatically use that `From` to convert the error](https://doc.rust-lang.org/stable/reference/expressions/operator-expr.html#the-question-mark-operator).

Reply:
Oh, and you might wonder why we have to do `Ok(thing_that_fails()?))`: I believe it's because while we get `From<EncryptError> for ClientError` for free, we don't get `from<Result<T, ClientError>> for Result<T, EncryptError>`, or at least `?` doesn't automagically invoke that conversion, so we have to do `?` to unwrap the `Result` from `self.encode_with`, and then wrap it back in `Ok` to be returned by this method.

Reply:
Damn, that's just magic. Done.